scripts/reanimated_utils.rb

```diff
@@ -17,15 +20,20 @@ def find_config()
     :react_native_common_dir => nil,
   }
 
-  react_native_node_modules_dir = File.join(File.dirname(`cd "#{Pod::Config.instance.installation_root.to_s}" && node --print "require.resolve('react-native/package.json')"`), '..')
-  react_native_json = try_to_parse_react_native_package_json(react_native_node_modules_dir)
+  # If the user set a custom location for where to look for React Native, use that
+  custom_react_native_node_modules_dir = ENV['REACT_NATIVE_NODE_MODULES_DIR']
+  if custom_react_native_node_modules_dir.nil?
+    react_native_node_modules_dir = File.join(File.dirname(`cd "#{Pod::Config.instance.installation_root.to_s}" && node --print "require.resolve('react-native/package.json')"`), '..')
+  else
+    unless File.exist? custom_react_native_node_modules_dir
+      raise "[RNReanimated] The given React Native lookup path #{custom_react_native_node_modules_dir} does not exist"
+    end
 
-  if react_native_json == nil
-    # user configuration, just in case
-    node_modules_dir = ENV["REACT_NATIVE_NODE_MODULES_DIR"]
```

Comment:
My rational for using `REACT_NATIVE_NODE_MODULES_DIR` _before_ trying to resolve the path ourselves is that if the users set a custom value, they likely expect it to be used.